jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/KModuleDeploymentService.java

```diff
@@ -293,23 +319,29 @@ protected Object getInstanceFromModel(ObjectModel model, KieContainer kieContain
     }
 
 	protected void processResources(InternalKieModule module, Collection<String> files,
-    		KieContainer kieContainer, DeploymentUnit unit, DeployedUnitImpl deployedUnit, ReleaseId releaseId) {
+    		KieContainer kieContainer, DeploymentUnit unit, DeployedUnitImpl deployedUnit, ReleaseId releaseId, Map<String, ProcessDescriptor> processes) {
         for (String fileName : files) {
             if(fileName.matches(".+bpmn[2]?$")) {
                 ProcessAssetDesc process;
                 try {
                     String processString = new String(module.getBytes(fileName), "UTF-8");
-                    process = (ProcessAssetDesc) bpmn2Service.buildProcessDefinition(unit.getIdentifier(), processString, kieContainer, true);
-                    if (process == null) {
-                    	throw new IllegalArgumentException("Unable to read process " + fileName);
+                    String processId = getProcessId(processString);
+                    ProcessDescriptor processDesriptor = processes.get(processId);
+                    if (processDesriptor != null) {
+                        process = processDesriptor.getProcess();
+                        if (process == null) {
+                            throw new IllegalArgumentException("Unable to read process " + fileName);
+                        }
+                        process.setEncodedProcessSource(Base64.encodeBase64String(processString.getBytes()));
+                        process.setDeploymentId(unit.getIdentifier());
+                        
+                        deployedUnit.addAssetLocation(process.getId(), process);
+                        bpmn2Service.addProcessDefinition(unit.getIdentifier(), processId, processDesriptor, kieContainer);
                     }
-                    process.setEncodedProcessSource(Base64.encodeBase64String(processString.getBytes()));
-                    process.setDeploymentId(unit.getIdentifier());
-                    deployedUnit.addAssetLocation(process.getId(), process);
                 } catch (UnsupportedEncodingException e) {
-                	throw new IllegalArgumentException("Unsupported encoding while processing process " + fileName);
+                    throw new IllegalArgumentException("Unsupported encoding while processing process " + fileName);
                 }
-            } else if (fileName.matches(".+ftl$") || fileName.matches(".+form$")) {
+            }if (fileName.matches(".+ftl$") || fileName.matches(".+form$")) {
```

Comment:
I think this `else if` should stay in here for readability (and to make absolutely sure that the file is processed correctly).